src/main/java/internet/shop/dao/impl/CarDaoImpl.java

```diff
@@ -1,54 +1,287 @@
+            throw new DataProcessingException("Can't update the data - "
+                    + car + "in table of cars or cars_drivers", exception);
+        }
+        createCarsDriversDB(car);
         return car;
     }
 
     @Override
     public boolean delete(Long id) {
-        return Storage.getCars().removeIf(i -> i.getId().equals(id));
+        String queryForCars = "UPDATE cars SET deleted = true where id = ?;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statementForCars = connection.prepareStatement(queryForCars)) {
+            statementForCars.setLong(1, id);
+            return statementForCars.executeUpdate() > 0;
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't delete the data by id: "
+                    + id + " into cars", exception);
+        }
     }
 
     public List<Car> getAllByDriver(Long driverId) {
-        return getAll().stream()
-                .filter(car -> car.getDrivers()
-                        .stream()
-                        .anyMatch(driver -> driver.getId().equals(driverId)))
-                .collect(Collectors.toList());
+        List<Car> carList = new ArrayList<>();
+        String query = "SELECT id from drivers where id = ? and deleted = false;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, driverId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                carList.addAll(getCarsByIdDrivers(connection,
+                        resultSet.getObject("id", Long.class)));
+            }
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't find drivers by id: " + driverId, exception);
+        }
+        return carList;
+    }
+
+    private void createCarsDriversDB(Car car) {
+        String queryForCarsDrivers = "INSERT INTO cars_drivers (driver_id, car_id) values (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatementOfCarsDrivers = connection
+                        .prepareStatement(queryForCarsDrivers)) {
+            Iterator<Driver> iterator = car.getDrivers().iterator();
+            while (iterator.hasNext()) {
+                preparedStatementOfCarsDrivers.setLong(1, iterator.next().getId());
+                preparedStatementOfCarsDrivers.setLong(2, car.getId());
+                preparedStatementOfCarsDrivers.executeUpdate();
+            }
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't create data "
+                    + car + " into cars_drivers table", exception);
+        }
+    }
+
+    private void removeCarsDrivers(Connection connection, Car car) {
```

Comment:
```suggestion
    private void removeCarsDrivers(Car car) {
```
open new connection inside this method

Reply:
НАпишу на русском. Блин, а я думал, что ты именно имелла ввиду за соеденение внутри методов. я думал нужно в других методах убрать с try-with-resources Connection и передавать как параметр. Всё теперь я понял точно